crypto/src/lib.rs

```diff
@@ -491,6 +506,12 @@ impl Default for Hash {
     }
 }
 
+impl CryptoHash for Hash {
+    fn hash(&self) -> Hash {
```

Comment:
it's probably a bit "strange" CryptoHash implementation. (It didn't cryptohash value)

Reply:
Yeah, @vldm, agree with you. But the reason of this implementation was in
```
error[E0119]: conflicting implementations of trait `storage::hash::UniqueHash` for type `crypto::Hash`:
  --> exonum/src/storage/hash.rs:35:1
   |
29 | impl<T: CryptoHash> UniqueHash for T {
   | ------------------------------------ first implementation here
...
35 | impl UniqueHash for Hash {
   | ^^^^^^^^^^^^^^^^^^^^^^^^ conflicting implementation for `crypto::Hash`
   |
   = note: upstream crates may add new impl of trait `crypto::CryptoHash` for type `crypto::Hash` in future versions
```
And I didn't get any better approach

Reply:
I've also found this weird and tried to change it. Without success though. Current solution looks not good, but maybe it's the best we can do.